chain-tx-enclave-next/enclave-ra/ra-client/src/verifier.rs

```diff
@@ -201,8 +229,8 @@ impl EnclaveCertVerifier {
         let quote = attestation_report_body.get_quote()?;
 
         if public_key.len() != 65
-            && public_key[0] != 4
-            && public_key[1..] != quote.report_body.report_data[..]
+            || public_key[0] != 4
+            || public_key[1..] != quote.report_body.report_data[..]
```

Comment:
a good catch!
It may be good to give names to the conditions, as there's currently no documentation e.g. referring to SECG.

```
let has_correct_len = public_key.len() == 65;
let is_uncompressed =  public_key[0] == 4;
let pubkey_matches = public_key[1..] == quote.report_body.report_data[..];
if ! has_correct_len || ! is_uncompressed || !pubkey_matches {
...
}

```